src/pywifes/extract_spec.py

```diff
@@ -0,0 +1,516 @@
+    red_var = red_cube_data["var"]
+    red_var_hdr = red_cube_data["var_hdr"]
+    red_wcs = red_cube_data["wcs"]
+    if red_cube_data["sci"] is not None:
+        binning_x = red_cube_data["binning_x"]
+        binning_y = red_cube_data["binning_y"]
+
+    # Calculate pixel scale from binning
+    pixel_scale_x = binning_x  # arcsec/pix
+    pixel_scale_y = binning_y / 2  # arcsec/pix
+
+    # Average all the data for detecting the source
+    collapsed_cube = collapse_cube(blue_sci, red_sci)
+
+    # Automatic source detection in the collapsed cubes (red + blue)
+    npeaks = 3  # Number of peaks to be detected
+    # Avoid the edges of size border_width on the statistics
+    collapsed_cube_no_edge = collapsed_cube[
+        border_width:-border_width, border_width:-border_width
+    ]
+    mean, median, std = sigma_clipped_stats(collapsed_cube_no_edge, sigma=5.0)
+    threshold = median + (3 * std)
+
+    detection = find_peaks(
+        collapsed_cube,
+        threshold,
+        border_width=border_width,
+        npeaks=npeaks,
+        centroid_func=centroid_com,
+    )
+
+    if detection is None:
+        print("No source detected.")
+
+    else:
+        # Sort detections as brighter peaks first
+        detection.sort("peak_value", reverse=True)
+
+        # Detected sources positions
+        positions = np.transpose((detection["x_peak"], detection["y_peak"]))
+
+        # Set the annulus
+        a = r_arcsec / pixel_scale_x
+        b = r_arcsec / pixel_scale_y
+
+        # Creates source apertures in the detected positions
+        source_apertures = EllipticalAperture(positions, a=a, b=b)
+
+        if sky_sub:
+            a_in = a * 3
+            a_out = a * 4
+
+            b_in = b * 3
+            b_out = b * 4
+
+            sky_aps = EllipticalAnnulus(
+                positions,
+                a_in=a_in,
+                a_out=a_out,
+                b_in=b_in,
+                b_out=b_out,
+            )
+
+        else:
+            sky_aps = None
+
+        # Flux extraction for the aperture at all the wavelenghts
+        for index, source_ap in enumerate(source_apertures):
+            if sky_aps is not None:
+                sky_ap = sky_aps[index]
+
+            else:
+                sky_ap = None
+            ap_index = index + 1
+
+            # Extracting blue cube
+            extract_and_save(
+                blue_cube_path,
+                blue_sci,
+                blue_var,
+                source_ap,
+                ap_index,
+                sky_ap,
+                output_dir,
+                blue_sci_hdr,
+                blue_var_hdr,
+            )
+
+            # Extracting red cube
+            extract_and_save(
+                red_cube_path,
+                red_sci,
+                red_var,
+                source_ap,
+                ap_index,
+                sky_ap,
+                output_dir,
+                red_sci_hdr,
+                red_var_hdr,
+            )
```

Comment:
Much easier to verify they're doing the same thing when you just have to compare the arguments, ty!